tests/test_doublyrobust.py

```diff
@@ -315,6 +335,266 @@ def test_sklearn_in_tmle_missing(self, mf):
+                                 [0, 0, 0, -1, 2],
+                                 [0, 1, 0, 5, 1],
+                                 [0, 0, 1, 0, 0],
+                                 [1, 0, 0, 0, 1],
+                                 [1, 0, 1, 0, 0],
+                                 [0, 1, 0, 10, 1],
+                                 [0, 0, 0, -5, 0],
+                                 [1, 1, 0, -5, 2]],
+                                columns=["W", "A", "Y", "C", "S"],
+                                index=[1, 2, 3, 4, 5, 6, 7, 8, 9])
+        return expected
+
+    def test_error_continuous_exp(self, df):
+        with pytest.raises(ValueError):
+            StochasticTMLE(df=df, exposure='cd40', outcome='dead')
+
+    def test_error_fit(self, df):
+        stmle = StochasticTMLE(df=df, exposure='art', outcome='dead')
+        with pytest.raises(ValueError):
+            stmle.fit(p=0.5)
+
+        stmle = StochasticTMLE(df=df, exposure='art', outcome='dead')
+        stmle.exposure_model('male + age0 + age_rs1 + age_rs2 + cd40 + cd4_rs1 + cd4_rs2 + dvl0')
+        with pytest.raises(ValueError):
+            stmle.fit(p=0.5)
+
+        stmle = StochasticTMLE(df=df, exposure='art', outcome='dead')
+        stmle.outcome_model('male + age0 + age_rs1 + age_rs2 + cd40 + cd4_rs1 + cd4_rs2 + dvl0')
+        with pytest.raises(ValueError):
+            stmle.fit(p=0.5)
+
+    def test_error_p_oob(self, df):
+        stmle = StochasticTMLE(df=df, exposure='art', outcome='dead')
+        stmle.exposure_model('male + age0 + age_rs1 + age_rs2 + cd40 + cd4_rs1 + cd4_rs2 + dvl0')
+        stmle.outcome_model('male + age0 + age_rs1 + age_rs2 + cd40 + cd4_rs1 + cd4_rs2 + dvl0')
+        with pytest.raises(ValueError):
+            stmle.fit(p=1.1)
+
+        with pytest.raises(ValueError):
+            stmle.fit(p=-0.1)
+
+    def test_error_p_cond_len(self, df):
+        stmle = StochasticTMLE(df=df, exposure='art', outcome='dead')
+        stmle.exposure_model('male + age0 + age_rs1 + age_rs2 + cd40 + cd4_rs1 + cd4_rs2 + dvl0')
+        stmle.outcome_model('male + age0 + age_rs1 + age_rs2 + cd40 + cd4_rs1 + cd4_rs2 + dvl0')
+        with pytest.raises(ValueError):
+            stmle.fit(p=[0.1], conditional=["df['male']==1", "df['male']==0"])
+
+        with pytest.raises(ValueError):
+            stmle.fit(p=[0.1, 0.3], conditional=["df['male']==1"])
+
+    def test_error_summary(self, df):
+        stmle = StochasticTMLE(df=df, exposure='art', outcome='dead')
+        stmle.exposure_model('male + age0 + age_rs1 + age_rs2 + cd40 + cd4_rs1 + cd4_rs2 + dvl0')
+        stmle.outcome_model('male + age0 + age_rs1 + age_rs2 + cd40 + cd4_rs1 + cd4_rs2 + dvl0')
+        with pytest.raises(ValueError):
+            stmle.summary()
+
+    def test_warn_missing_data(self):
+        df = pd.DataFrame()
+        df['A'] = [1, 1, 0, 0, np.nan]
+        df['Y'] = [np.nan, 0, 1, 0, 1]
+        with pytest.warns(UserWarning):
```

Comment:
ditto above comment. This could start "passing" from some warning thrown in a dependent library.